src/ml_models.py

```diff
@@ -8,11 +8,14 @@
 import os
 import tempfile
 from dataclasses import dataclass
-from typing import Callable, Iterator, List, Optional, Sequence, Tuple
+from typing import Any, Callable, Iterator, List, Optional, Sequence, Tuple
 
 import numpy as np
 import torch
 import torch.nn.functional as F
+from sklearn.base import BaseEstimator
+from sklearn.neighbors import KNeighborsClassifier as _KNeighborsClassifier
```

Comment:
better naming for these might be `SKLearnKNeighborsClassifier` and `SKLearnKNeighborsRegressor`